privacy-notices-updater/src/main/scala/it/pagopa/interop/privacynoticesupdater/service/DynamoService.scala

```diff
@@ -0,0 +1,58 @@
+package it.pagopa.interop.privacynoticesupdater.service
+
+import cats.syntax.all._
+import com.typesafe.scalalogging.{Logger, LoggerTakingImplicit}
+import it.pagopa.interop.commons.logging.{CanLogContextFields, ContextFieldsToLog}
+import it.pagopa.interop.commons.utils.TypeConversions._
+import it.pagopa.interop.privacynoticesupdater.util.DynamoConfiguration
+import it.pagopa.interop.privacynoticesupdater.model.db.PrivacyNotice
+import it.pagopa.interop.privacynoticesupdater.error.PrivacyNoticeError._
+import org.scanamo.DynamoReadError.describe
+import org.scanamo._
+import org.scanamo.syntax._
+
+import scala.concurrent.{Future, ExecutionContext}
+import java.util.UUID
+
+trait DynamoService {
+  def getById(id: UUID)(implicit contexts: Seq[(String, String)]): Future[Option[PrivacyNotice]]
+
+  def put(privacyNotice: PrivacyNotice)(implicit contexts: Seq[(String, String)]): Future[Unit]
+
+  def delete(id: UUID)(implicit contexts: Seq[(String, String)]): Future[Unit]
+}
+
+final class DynamoServiceImpl(config: DynamoConfiguration)(implicit ec: ExecutionContext, scanamo: ScanamoAsync)
+    extends DynamoService {
+
+  implicit val logger: LoggerTakingImplicit[ContextFieldsToLog] =
+    Logger.takingImplicit[ContextFieldsToLog](this.getClass)
+
+  val table: Table[PrivacyNotice] =
+    Table[PrivacyNotice](config.tableName)
+
+  override def put(privacyNotice: PrivacyNotice)(implicit contexts: Seq[(String, String)]): Future[Unit] = {
+    logger.info(s"Putting $privacyNotice privacy notice")
+    scanamo
+      .exec(table.put(privacyNotice))
+  }
+
+  override def delete(id: UUID)(implicit contexts: Seq[(String, String)]): Future[Unit] = {
+    logger.info(s"Deleting privacy notice with id $id")
+    scanamo
+      .exec(
+        table
+          .delete("pk" === s"${PrivacyNotice.pkPrefix}$id" and "sk" === s"${PrivacyNotice.skPrefix}$id")
+      )
```

Comment:
![hadouken](https://github.com/pagopa/interop-be-jobs/assets/41690956/aa00193c-adba-4cb5-beef-1141eaf476ba)

Reply:
Have I to rewrite this without using Free Monad ScanamoOps?